decorators.py

```diff
@@ -1,8 +1,39 @@
+import copy
 from datetime import date
 from functools import wraps
+
 from flask import request, g, render_template
 
 
+def get_permissible_entries(user, entry):
```

Comment:
I'd love to see some unittests that show that the code is working as intended. Any GSoC participant working on this project will have to do that anyway, so it's best to get used to it already :+1:

Reply:
I'm working on it, Will update you soon :)

Reply:
Unittests have been added in latest commits.